modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/build.gradle

```diff
@@ -39,6 +39,10 @@ dependencies {
 	compileInclude group: "commons-io", name: "commons-io", version: "2.11.0"
 	compileInclude group: "commons-lang", name: "commons-lang", version: "2.6"
 	compileInclude group: "org.apache.commons", name: "commons-compress", version: "1.21"
+	compileInclude group: "org.apache.httpcomponents", name: "httpasyncclient", version: "4.1.4"
```

Comment:
@AlmirFe im curious what is the reason for switching from compileOnly to compileInclude?

Reply:
Hi @BryanEngler,

After changing the version of the elasticsearch client, the error below started to appear in the log whenever I tried to deploy the changes:

```
Unresolved requirement: Import-Package: org.apache.http.nio.client_ [Sanitized]
	at org.eclipse.osgi.container.Module.start(Module.java:444) ~[org.eclipse.osgi.jar:?]
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:426) ~[org.eclipse.osgi.jar:?]
	at com.liferay.portal.file.install.internal.DirectoryWatcher._startBundle(DirectoryWatcher.java:1147) [bundleFile:?]
	at com.liferay.portal.file.install.internal.DirectoryWatcher._startBundles(DirectoryWatcher.java:1180) [bundleFile:?]
	at com.liferay.portal.file.install.internal.DirectoryWatcher._startAllBundles(DirectoryWatcher.java:1121) [bundleFile:?]
	at com.liferay.portal.file.install.internal.DirectoryWatcher._process(DirectoryWatcher.java:1032) [bundleFile:?]
	at com.liferay.portal.file.install.internal.DirectoryWatcher.run(DirectoryWatcher.java:238) [bundleFile:?]
```

The **org.apache.http.nio.client** dependency is part of the "org.apache.httpcomponents" group which, as you can see, was already being declared in the build.gradle file. This error was probably being caused by a conflict in dependencies with another component that uses the same dependencies. To solve this problem, I replaced compileOnly with compileInclude. This way, gradlew includes these dependencies within the generated jar, which eliminates any possibility of conflict.

Reply:
Sometimes, instead of adding, excluding the problematic packages through the `bnd.bnd`. also helps. :thinking:

Reply:
interesting, i didnt see any of these errors... maybe we can get another person to try it out

Reply:
elastic doesnt "support" hot re-deploys. you have to stop and start the server... maybe that was part of the problem you saw

Reply:
The case I talked about was about deploying and restarting the portal. The error I showed earlier was not caused by this.

Reply:
The case I talked about was about deploying and restarting the portal. The error I showed earlier was not caused by this.

Reply:
running ci with compileOnly on this pr https://github.com/liferay-search/liferay-portal/pull/1128 to see what happens

Reply:
Im looing at the commits for https://liferay.atlassian.net/browse/LPS-168749, makes me think that we should be able to keep it compileOnly... im gonna reach out to Jeff Wu

Reply:
Jeff gave me some info, testing on https://github.com/liferay-search/liferay-portal/pull/1128